### PR TITLE
Ignore empty List<String> configuration values

### PR DESCRIPTION
**Pull request by chonton**
Allow empty list elements.  This often occurs when a parent (corporate) pom has configuration values that are overridden by some, but not all, children.

### CODE REVIEW

File: src/main/java/io/fabric8/maven/docker/config/BuildImageConfiguration.java [hunk 1]
```diff
@@ -175,16 +177,19 @@ public AssemblyConfiguration getAssemblyConfiguration() {
         return assembly;
     }
 
+    @Nonnull
     public List<String> getPorts() {
-        return ports;
+        return EnvUtil.removeEmpties(ports);
```
Comment (rhuss) [inline src/main/java/io/fabric8/maven/docker/config/BuildImageConfiguration.java]: minor nitpick: Could we rename it to removeNullEntries() ?

Comment (chonton) [inline src/main/java/io/fabric8/maven/docker/config/BuildImageConfiguration.java]: Sure, however it removes both null and empty values.

Comment (rhuss) [inline src/main/java/io/fabric8/maven/docker/config/BuildImageConfiguration.java]: Ah. ok. I thought it was about null entries. We should keep it of course like this, maybe calling it 'removeEmptyEntries()' ?

Trim is find for me, too.

**Comment by chonton-elementum** [comment]
@rhuss, did you intend to merge this into master or not?  Do I need to create another pull request with new base?

**Comment by rhuss** [comment]
Sorry, I simply forgot to merge ;-)

[merge]

**Pull request merged**